### PR TITLE
Non-throwing is_identifier_[start_]char

### PR DESCRIPTION
**Pull request by pfitzseb**

### CODE REVIEW

File: src/tokenize_utils.jl [hunk 1]
```diff
@@ -4,12 +4,21 @@ const EOF_CHAR = typemax(Char)
 
 function is_identifier_char(c::Char)
     c == EOF_CHAR && return false
-    return Base.is_id_char(c)
+    return try
+        Base.is_id_char(c)
+    catch _
```
[the remark this answers is not in the document]
Comment (pfitzseb) [reply src/tokenize_utils.jl]: Sure, why not. Looks weird to me though :P

File: src/tokenize_utils.jl [hunk 2]
```diff
@@ -4,12 +4,21 @@ const EOF_CHAR = typemax(Char)
 
 function is_identifier_char(c::Char)
     c == EOF_CHAR && return false
-    return Base.is_id_char(c)
+    return try
```
Comment (c42f) [inline src/tokenize_utils.jl:7]: Might be faster to use `Base.ismalformed()` here as a pre-check rather than the `try ... catch` ?

**Review by c42f — COMMENTED** [review]
Thanks!  Could do with a test case.

**Review by c42f — APPROVED** [review]
looks great

**Pull request merged**